codegen/codegen.c

```diff
@@ -15,6 +15,8 @@ context new_context()
     ctx.symbol_table = new_linked_list();
     ctx.structs = new_linked_list();
     ctx.label_counter = 0;
+    ctx.current_loop_end_label_index = 0;
```

Comment:
In case you need an stack you may use the `linked_list` data structure. I think we already have a pop function